### PR TITLE
update - istio add standard labels

### PR DESCRIPTION
**Pull request by lachie83**
* add standard labels
* Fix conditional on ca deployment
* Bump chart version
* Update istio labels
* use serviceName across all manifests

### CODE REVIEW

**Comment by lachie83** [comment]
cc @unguiculus

File: incubator/istio/templates/_helpers.tpl [hunk 1]
```diff
@@ -14,3 +14,29 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
...
 {{- end -}}
+
+{{- /*
+Credit: @technosophos
+https://github.com/technosophos/common-chart/
+labels.standard prints the standard Helm labels.
+The standard labels are frequently used in metadata.
+*/ -}}
+{{- define "labels.standard" -}}
+app: {{template "fullname" .}}
```
Comment (unguiculus) [inline incubator/istio/templates/_helpers.tpl]: Let's use `{{ template "name" . }}` for the `app` label. See recent discussion on Slack about this. Note that you will have to update label selectors in services. See e. g. here: https://github.com/kubernetes/charts/blob/master/stable/nginx-ingress/templates/controller-service.yaml#L55

And please use consistent formatting everywhere, i. e. space after `{{` and space before `}}`.

Apply everywhere in this chart.

File: incubator/istio/templates/addons-prometheus-svc.yaml [hunk 2]
```diff
@@ -3,15 +3,15 @@
 apiVersion: v1
 kind: Service
 metadata:
+  # Grafana datasource config is hardcoded to prometheus:9090
```
Comment (unguiculus) [inline incubator/istio/templates/addons-prometheus-svc.yaml:6]: Does it have to be?

**Review by unguiculus — CHANGES_REQUESTED** [review]
I think it's great to use a template function for standard labels. Can we include this in the default `_helpers.tpl`, so everyone gets it with `helm create`? Currently, `helm create` doesn't even add standard labels at all.

/cc @technosophos

File: incubator/istio/templates/addons-prometheus-svc.yaml (hunk 2 above)
Comment (lachie83) [reply incubator/istio/templates/addons-prometheus-svc.yaml:6]: Currently yes.

File: incubator/istio/templates/_helpers.tpl (hunk 1 above)
Comment (lachie83) [reply incubator/istio/templates/_helpers.tpl]: Agreed

**Comment by lachie83** [comment]
@unguiculus Updated with your changes. Please review

**Comment by lachie83** [comment]
@k8s-bot pull-charts-e2e test this

File: incubator/istio/templates/_helpers.tpl [hunk 3]
```diff
@@ -14,3 +14,29 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
...
+Credit: @technosophos
+https://github.com/technosophos/common-chart/
+labels.standard prints the standard Helm labels.
+The standard labels are frequently used in metadata.
+*/ -}}
+{{- define "labels.standard" -}}
+app: {{template "name" . }}
+heritage: {{ .Release.Service | quote }}
+release: {{ .Release.Name | quote }}
+chart: {{template "chartref" . }}
```
Comment (unguiculus) [inline incubator/istio/templates/_helpers.tpl]: Still some missing spaces here.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Thanks for the update. We're not quite there yet. I would add `component` annotations for the individual components. Look at how @mgoodness did it. I think that's the way to go.

https://github.com/kubernetes/charts/blob/master/stable/nginx-ingress/templates/controller-service.yaml#L8-L14

You will also have to change label selectors in the services:

https://github.com/kubernetes/charts/blob/master/stable/nginx-ingress/templates/controller-service.yaml#L55-L58

**Comment by lachie83** [comment]
Updated based on @unguiculus feedback

File: incubator/istio/values.yaml [hunk 4]
```diff
@@ -191,6 +200,7 @@ addons:
       externalPort: 3000
     
     deployment:
+      name: grafan
```
Comment (unguiculus) [inline incubator/istio/values.yaml]: grafana?

**Review by unguiculus — DISMISSED** [review]
One more little thing...

**Comment by lachie83** [comment]
fixed thanks! 👍 @unguiculus

**Comment by unguiculus** [comment]
lgtm

**Pull request merged**